File: app/src/main/java/com/kota/hh/injection/module/ApplicationModule.java
package com.kota.hh.injection.module;

import android.app.Application;
import android.content.Context;

import com.kota.hh.data.remote.ApiService;
import com.kota.hh.injection.ApplicationContext;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Provide application-level dependencies.
 */

@Module
public class ApplicationModule {
    protected final Application mApplication;

    public ApplicationModule(Application application) {
        mApplication = application;
    }

    @Provides
    Application provideApplication() {
        return mApplication;
    }

    @Provides
    @ApplicationContext
    Context provideContext() {
        return mApplication;
    }

    @Provides
    @Singleton
    ApiService provideApiService() {
        return ApiService.Companion.newApiService();
    }

}

File: app/src/main/java/com/kota/hh/injection/module/ApplicationModule.kt
package com.kota.hh.injection.module

import android.app.Application
import android.content.Context
import com.kota.hh.data.remote.ApiService
import com.kota.hh.injection.ApplicationContext
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ApplicationModule(val application: Application) {

    @Provides
    fun provideApplication() = application

    @Provides
    @ApplicationContext
    fun provideContext(): Context = application

    @Provides
    @Singleton
    fun provideApiService() = ApiService.newApiService()
}